src/tests/test_sqlthread.py

```diff
@@ -0,0 +1,77 @@
+"""
+    Test for sqlThread
+"""
+
+import os
+import unittest
+import sqlite3
+import sys
+from ..state import appdata
+from ..helper_sql import sqlStoredProcedure, sql_ready, executeCF
+from ..class_sqlThread import (sqlThread)
+from ..addresses import encodeAddress
+
+
+class TestSqlThread(unittest.TestCase):
+    """
+        Test case for SQLThread
+    """
+
+    conn = sqlite3.connect(appdata + 'messages.dat')
+    conn.text_factory = str
+    cur = conn.cursor()
+
+    # query file path
+    root_path = os.path.dirname(os.path.dirname(__file__))
+
+    @classmethod
+    def setUpClass(cls):
+        # Start SQL thread
+        sqlLookup = sqlThread()
+        sqlLookup.daemon = False
+        sqlLookup.start()
+        sql_ready.wait()
+
+    @classmethod
+    def setUp(cls):
+        tables = list(cls.cur.execute("select name from sqlite_master where type is 'table'"))
+        cls.cur.executescript(';'.join(["drop table if exists %s" % i for i in tables]))
+
+    @classmethod
+    def tearDown(cls):
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        # Stop sql thread
+        sqlStoredProcedure('exit')
+
+    def initialise_database(self, file):
+        """
+            Initialise DB
+        """
+
+        sql_file_path = os.path.join(self.root_path, "tests/sql/{}.sql".format(file))
+        sql_file = open(sql_file_path)
+        sql_as_string = sql_file.read()
+        self.cur.executescript(sql_as_string)
+        self.conn.commit()
+
+    def test_create_function(self):
+        # call create function
+        executeCF(self.conn)
+
+        encoded_str = encodeAddress(1, 1, "21122112211221122112")
```

Comment:
these should be class variables or constants defined elsewhere.

Also version should be 4.